### PR TITLE
refactor(dev-env): let `dev-tools` handle their volumes

### PR DESCRIPTION
**Pull request by sjinks**
## Description

This PR updates the dev-env code to avoid messing with the `dev-tools` volume.

1. We no longer remove `dev-tools` volumes;
2. We ensure that the `php` service starts after `devtools` completes to avoid race conditions;
3. We sync `/dev-tools-orig` with `/dev-tools`. This step is necessary because Lando sets its own entrypoint script if we don't provide ours.

Related PR: Automattic/vip-container-images#985 (must be merged before this one)
Volume removal logic was introduced in #1317.

## Pull request checklist

- [ ] Update [SETUP.md](https://github.com/Automattic/vip-cli/blob/trunk/docs/SETUP.md#list-of-environmental-variables) with any new environmental variables.
- [ ] Update [the documentation](https://github.com/Automattic/vip-cli/blob/trunk/docs).
- [x] [Manually test](https://github.com/Automattic/vip-cli/blob/trunk/docs/TESTING.md#manual-testing) the relevant changes.
- [x] Follow the [pull request checklist](https://github.com/Automattic/vip-cli/blob/trunk/docs/RELEASING.md#new-pull-requests)
- [ ] Add/update [automated tests](https://github.com/Automattic/vip-cli/blob/trunk/docs/TESTING.md#automated-testing) as needed.

## New release checklist

- [ ] [Automated tests](https://github.com/Automattic/vip-cli/blob/trunk/docs/TESTING.md#automated-testing) pass.
- [ ] The [Preparing for release checklist](https://github.com/Automattic/vip-cli/blob/trunk/docs/RELEASING.md#preparing-for-release) is completed.

## Steps to Test

1. Check out Automattic/vip-container-images#985, `cd dev-tools && docker build -t dev-tools:local`
2. Check out this PR, `npm run build && npm link`
3. vip dev-env create < /dev/null
4. Create `~/.local/share/vip/dev-environment/vip-local/.lando.local.yml`:
```yml
services:
  devtools:
    services:
      image: dev-tools:local
      pull_policy: never
```
5. Run `vip dev-env start` and ensure everything works
6. Run `vip dev-env shell -- sh -c 'md5sum /dev-tools/*'` and save the output.
7. Update one of the dev-tools scripts and rebuild the image (`docker build -t dev-tools:local`)
8. Restart the environment (`vip dev-env start`).
9. Run `vip dev-env shell -- sh -c 'md5sum /dev-tools/*'` and compare the output with the one from step 6. The hash for the updated script must differ.

### CODE REVIEW

File: assets/dev-env.lando.template.yml.ejs [hunk 1]
```diff
@@ -31,6 +30,7 @@ services:
       devtools: {}
       scripts:
     initOnly: true
+    entrypoint: sh -c 'test -d /dev-tools-orig && /usr/bin/rsync -a --delete /dev-tools-orig/ /dev-tools/ || true'
```
Comment (sjinks) [inline assets/dev-env.lando.template.yml.ejs:29]: Backward compatibility :-(

**Comment by rinatkhaziev** [comment]
I've tested both for a) existing environment and b) fresh env under Mac OS X
1. The good news is that files do get updated on image rebuild. 
2. The bad news that on environment start `nginx` quits with

```
2024-12-06 11:15:25 lando 17:15:25.INFO  ==> Lando handing off to: nginx -g daemon off;
2024-12-06 11:15:25 lando 17:15:25.DEBUG ==> Running command with exec...
2024-12-06 11:15:25 2024/12/06 17:15:25 [emerg] 1#1: host not found in upstream "php" in /etc/nginx/conf.d/default.conf:60
2024-12-06 11:15:25 nginx: [emerg] host not found in upstream "php" in /etc/nginx/conf.d/default.conf:60
```

Manual start of nginx container after that works.

**Comment by sjinks** [comment]
Updated service dependencies.

**Comment by rinatkhaziev** [comment]
Tested, works!

**Review by rinatkhaziev — APPROVED** [review]

**Pull request merged**